suitcase/hdf5.py

```diff
@@ -35,26 +46,36 @@ def export(headers, filename):
             for i, descriptor in enumerate(descriptors):
                 # make sure it's a dictionary and trim any spurious keys
                 descriptor = dict(descriptor)
+                if stream_name:
+                    if descriptor['name'] != stream_name:
+                        continue
                 descriptor.pop('_name', None)
 
                 desc_group = group.create_group(descriptor['uid'])
 
-                data_keys = descriptor.pop('data_keys')
+                data_keys = dict(descriptor.pop('data_keys'))
+                if fields_unwanted is not None:
+                    for key in fields_unwanted:
+                        data_keys.pop(key, None)
+
                 _safe_attrs_assignment(desc_group, descriptor)
 
                 events = list(get_events_generator(descriptor=descriptor))
                 event_times = [e['time'] for e in events]
                 desc_group.create_dataset('time', data=event_times,
                                           compression='gzip', fletcher32=True)
                 data_group = desc_group.create_group('data')
-                ts_group = desc_group.create_group('timestamps')
+                if timestamps_opt:
+                    ts_group = desc_group.create_group('timestamps')
                 [fill_event(e) for e in events]
+
                 for key, value in data_keys.items():
                     value = dict(value)
```

Comment:
Not sure this `dict()` is necessary either (though this one isn't your fault :grinning: )